src/rsz/src/EstimateWireParasitics.cc

```diff
@@ -273,17 +274,37 @@ void Resizer::ensureParasitics()
 
 void Resizer::estimateParasitics(ParasiticsSrc src)
 {
+  std::map<Corner*, std::ostream*> spef_streams;
+  estimateParasitics(src, spef_streams);
+}
+
+void Resizer::estimateParasitics(ParasiticsSrc src,
+                                 std::map<Corner*, std::ostream*>& spef_streams)
+{
+  if (!spef_streams.empty()) {
+    spef_writer_ = new SpefWriter(logger_, sta_, spef_streams);
+    for (auto& [corner, _] : spef_streams) {
+      spef_writer_->writeSpefHeader(corner);
+      spef_writer_->writeSpefPorts(corner);
+    }
+  }
+
   switch (src) {
     case ParasiticsSrc::placement:
       estimateWireParasitics();
```

Comment:
In a future PR it would be good to handle placement based parasitics too.

Reply:
It was done by setting a Resizer spef_writer_ member instead of passing the SpefWriter* through functions like in GlobalRouter. I've changed it to be similar to GlobalRouter now.

I'll make a new commit after we end the discussion on the other comments.